torch/csrc/inductor/aot_inductor_tensor.h

```diff
@@ -0,0 +1,80 @@
+#ifndef __AOT_INDUCTOR_TENSOR__
+#define __AOT_INDUCTOR_TENSOR__
+
+#include <stdint.h>
+
+#define EXPORT __attribute__((__visibility__("default")))
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+typedef enum {
+  kAotInductorCPU = 1,
+  kAotInductorCUDA = 2,
+} AotInductorDeviceType;
+
+typedef struct {
+  AotInductorDeviceType device_type;
+  int8_t device_id;
+} AotInductorDevice;
+
+typedef enum {
+  kAotInductorByte = 1,
+  kAotInductorChar = 2,
+  kAotInductorShort = 3,
+  kAotInductorInt = 4,
+  kAotInductorLong = 5,
+  kAotInductorHalf = 6,
+  kAotInductorFloat = 7,
+  kAotInductorDouble = 8,
+  kAotInductorComplexHalf = 9,
+  kAotInductorComplexFloat = 10,
+  kAotInductorComplexDouble = 11,
+  kAotInductorBool = 12,
+  kAotInductorBFloat16 = 13,
+} AotInductorScalarType;
+
+// Bare minumum tensor struct to interchange data between AOTInducotor
+// generated shared library and libtorch to avoid ABI compatibility issues
+typedef struct {
+  void* data_ptr;
+  AotInductorDevice device;
+  AotInductorScalarType type;
+  int64_t ndim;
+  const int64_t* sizes;
+  const int64_t* strides;
+} AotInductorTensor;
+
+EXPORT AotInductorTensor convert_to_aot_inductor_tensor(void* aten_tensor);
+
+EXPORT void convert_to_aten_tensor(
+    AotInductorTensor inductor_tensor,
+    void* aten_tensor);
+
+EXPORT AotInductorTensor aot_inductor_empty_strided(
+    int64_t dim,
+    int64_t* sizes_ptr,
+    int64_t* strides_ptr,
+    AotInductorDevice device,
+    AotInductorScalarType type);
+
+EXPORT AotInductorTensor aot_inductor_as_strided(
```

Comment:
You may want to cherry-pick the `torch.ops.inductor._reinterpret_tensor` part of #102625 and use that instead of burning the name `as_strided` into this API.

There is actually a subtle bug in how TorchInductor uses `as_strided`.  The difference between the two is:
- as_strided discards the storage_offeset() of the input tensor
- _reinterpret_tensor adds the new offset to the storage_offeset() of the input tensor

So they act the same on a tensor with a `storage_offset=0`, but differ when `storage_offset>0`.

Basically whenever inductor generates as_strided, it really means _reinterpret_tensor.  It should be possible to construct a test case that triggers this bug, by chaining offsets with more complex views.

Currently, this doesn't bite us in practice, because usually `storage_offset=0`.  However, when we start doing memory planning it will become an issue.